tests/routes/test_admin.py

```diff
@@ -512,3 +561,96 @@ def test_bulk_import_cclw_law_policy_preexisting_db_objects(
     mock_write_csv_to_s3.assert_called_once()
     call = mock_write_csv_to_s3.mock_calls[0]
     assert len(call.kwargs["file_contents"]) == csv_file.getbuffer().nbytes
+
+    mock_update_doc_in_db.assert_called_once()
+    mock_delete_doc_in_s3.assert_called_once()
+
+
+def test_bulk_import_cclw_law_policy_document_updates(
+    client,
+    superuser_token_headers,
+    test_db,
+    mocker,
+):
+    """
+    Test that the document update feature of the bulk import endpoint works as expected.
+
+    The test:
+    - Inserts the relevant metadata into the database.
+    - Bulk imports an initial csv file.
+    - Bulk imports a csv file with an update to a document.
+    - Asserts that the document was updated in the database.
+    """
+    mock_update_doc_in_db = mocker.patch(
+        "app.api.api_v1.routers.admin.update_doc_in_db"
+    )
+    mock_delete_doc_in_s3 = mocker.patch(
+        "app.api.api_v1.routers.admin.delete_doc_in_s3"
+    )
+
+    test_db.add(Source(name="CCLW"))
+    test_db.add(
+        Geography(
+            display_value="geography", slug="geography", value="GEO", type="country"
+        )
+    )
+    test_db.add(DocumentType(name="doctype", description="doctype"))
+    test_db.add(Language(language_code="LAN", name="language"))
+    test_db.add(Category(name="Policy", description="policy"))
+    test_db.add(Keyword(name="keyword1", description="keyword1"))
+    test_db.add(Keyword(name="keyword2", description="keyword2"))
+    test_db.add(Hazard(name="hazard1", description="hazard1"))
+    test_db.add(Hazard(name="hazard2", description="hazard2"))
+    test_db.add(Response(name="topic", description="topic"))
+    test_db.add(Framework(name="framework", description="framework"))
+
+    test_db.commit()
+
+    test_db.add(Instrument(name="instrument", description="instrument", source_id=1))
+    test_db.add(Sector(name="sector", description="sector", source_id=1))
+
+    test_db.commit()
+
+    csv_file = BytesIO(VALID_FILE_1.encode("utf8"))
+    files = {"law_policy_csv": ("valid.csv", csv_file, "text/csv", {"Expires": "0"})}
+    response = client.post(
+        "/api/v1/admin/bulk-imports/cclw/law-policy",
+        files=files,
+        headers=superuser_token_headers,
+    )
+    assert response.status_code == 202
+    response_json = response.json()
+    assert response_json["document_count"] == 2
+    assert response_json["document_updated_count"] == 0
+    assert response_json["document_skipped_count"] == 0
```

Comment:
Just realised we are returning two truths here: `document_skipped_count` and `len(document_skipped_ids)`